karabo_data/geometry2.py

```diff
@@ -1384,14 +1384,15 @@ def to_distortion_array(self):
             indexing='ij'
         )
         # Every second line of pixels across the slow-scan direction is shifted
-        # half a pixel in the fast-scan direction so the hexagons tessalate.
-        # TODO: check we're shifting the right way once we know the pixel order
-        pixel_fs_index[1::2, :] += 0.5
+        # half a pixel against the fast-scan direction so the hexagons tessalate.
+        # TODO: double check this shifting is correct
+        pixel_fs_index[1::2, :] -= 0.5
 
         # Corners described clockwise from the top, assuming the reference point
         # for a pixel is outside it, aligned with the top point & left edge.
-        corner_ss_offsets = np.array([0.5, 1, 1, 0.5, 0, 0])
-        corner_fs_offsets = np.array([0, 0.25, 0.75, 1, 0.75, 0.25])
+        # The 4/3 extends the hexagons into the next row to correctly tessellate.
+        corner_ss_offsets = np.array([0, 0.25, 0.75, 1, 0.75, 0.25]) * 4 / 3
+        corner_fs_offsets = np.array([0.5, 1, 1, 0.5, 0, 0])
```

Comment:
non expert in pyFAI here: Since the hexagons are stretched (204/236 I think), should this impact these corner positions? Or 1 unit in one axis does not necessarily equals 1 unit in the other axis?

Also is the fast scan the longest axis for DSSC?

Reply:
The 204/236 stretch is taken care of elsewhere - that's effectively the aspect ratio to get all pixels in (nearly) the right position.

But the 204 μm vertical distance is the step from the centre of one row of pixels to the centre of the next. If we just use that (i.e. remove the 4/3), then the top points of hexagons in row 0 are in line with the bottom points of row 1. We want them to interlock like the teeth of a gear. So the 4/3 is the factor between 204 and 272:

![Pixel_dims](https://user-images.githubusercontent.com/327925/59775940-e627b100-92a9-11e9-83fc-889605b2f042.png)

I did also check that 4/3 is what I'd expect for ideal regular hexagons.

The long axis (512 pixels) is indeed fast-scan for DSSC.

Reply:
aha, I understand. Thanks for explaining.